### PR TITLE
Feat/introduce action evaluation publisher

### PR DESCRIPTION
**Pull request by area363**
SSIA

### CODE REVIEW

File: NineChronicles.DataProvider/ActionEvaluationPublisher.cs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        private RpcContext _context;
+
+        public ActionEvaluationPublisher(
+            BlockRenderer blockRenderer,
+            ActionRenderer actionRenderer,
+            ExceptionRenderer exceptionRenderer,
+            NodeStatusRenderer nodeStatusRenderer,
+            string host,
+            int port,
+            RpcContext context
```
Comment (longfin) [inline NineChronicles.DataProvider/ActionEvaluationPublisher.cs:36]: These arguments don't seem necessary.

File: NineChronicles.DataProvider/ActionEvaluationPublisher.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            _blockRenderer = blockRenderer;
+            _actionRenderer = actionRenderer;
+            _exceptionRenderer = exceptionRenderer;
+            _nodeStatusRenderer = nodeStatusRenderer;
+            _host = host;
+            _port = port;
+            _context = context;
+        }
+
+        public override async Task StartAsync(CancellationToken stoppingToken)
```
Comment (longfin) [inline NineChronicles.DataProvider/ActionEvaluationPublisher.cs:48]: This method also seems unncessary.

File: NineChronicles.DataProvider/ActionEvaluationPublisher.cs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                        Log.Error(e, "Skip broadcasting status change due to the unexpected exception");
+                    }
+                },
+                stoppingToken
+            );
+
+            return Task.CompletedTask;
+        }
+
+        private bool ContainsAddressToBroadcast(ActionBase.ActionEvaluation<ActionBase> ev)
```
Comment (longfin) [inline NineChronicles.DataProvider/ActionEvaluationPublisher.cs:194]: It seems unnecessary.

File: NineChronicles.DataProvider/ActionEvaluationPublisher.cs [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        }
+
+        public override async Task StartAsync(CancellationToken stoppingToken)
+        {
+            await base.StartAsync(stoppingToken);
+        }
+
+        protected override Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            _blockRenderer.EveryBlock().Subscribe(
```
Comment (longfin) [inline NineChronicles.DataProvider/ActionEvaluationPublisher.cs:55]: How about clear subscriptions and leave only `_actionRenderer.EveryRender<ActionBase>()`?

**Comment by area363** [comment]
PTAL @longfin :pray:

**Review by longfin — APPROVED** [review]

**Pull request merged**